registry/server/tag_cleanup.go

```diff
@@ -100,19 +118,22 @@ func (s *Server) DeleteStaleTags(now func() time.Time, c Config) {
 
 		markTime, err := strconv.Atoi(markTag)
 		if err != nil {
-			log.Printf("Found non timestamp tag %s in stale tag marker\n", markTag)
+			log.Printf("found non timestamp tag %s in stale tag marker\n", markTag)
 		}
 
 		if sweepTime-int64(markTime) > int64(c.TagAllowedStalenessMinutes*60) {
-			keys := make([]string, len(tags))
-			i := 0
-			for k := range tags {
-				keys[i] = k
-				i++
-			}
+			// This object has tags being deleted; add it to the set.
+			deletedObjects[kafkaObject] = struct{}{}
+
+			keys := tags.Keys()
 			s.Tags.Store.DeleteTags(kafkaObject, keys)
 		}
 	}
+
+	// Log names of resources that have had tags deleted.
+	for object := range deletedObjects {
+		log.Printf("deleted tags for non-existent %s %s\n", object.Type, object.ID)
```

Comment:
Why log at the end instead of inline?

Reply:
Good Q, I structured it that way when we did the initial live debugging but it doesn't seem to make sense atm - going to update that one